timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java

```diff
@@ -0,0 +1,174 @@
+import com.palantir.timelock.history.models.AcceptorUseCase;
+import com.palantir.timelock.history.models.LearnerUseCase;
+import com.palantir.timelock.history.remote.TimeLockPaxosHistoryProviderResource;
+import com.palantir.timelock.history.sqlite.SqlitePaxosStateLogHistory;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+import org.immutables.value.Value;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public final class TimeLockCorruptionTestSetup implements TestRule {
+    private TemporaryFolder tempFolder = new TemporaryFolder();
+    private DataSource localDataSource;
+    private DataSource remoteDataSource1;
+    private DataSource remoteDataSource2;
+    private PaxosLogHistoryProvider paxosLogHistoryProvider;
+
+    private StateLogComponents defaultLocalServer;
+    private List<StateLogComponents> defaultRemoteServerList;
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return RuleChain.outerRule(tempFolder)
+                .around(new ExternalResource() {
+                    @Override
+                    protected void before() {
+                        try {
+                            setup();
+                        } catch (Throwable throwable) {
+                            // no op
+                        }
+                    }
+
+                    @Override
+                    protected void after() {
+                        // no op
+                    }
+                })
+                .apply(base, description);
+    }
+
+    private void setup() throws Throwable {
+        localDataSource = SqliteConnections.getPooledDataSource(
+                tempFolder.newFolder("randomFile1").toPath());
+        remoteDataSource1 = SqliteConnections.getPooledDataSource(
+                tempFolder.newFolder("randomFile2").toPath());
+        remoteDataSource2 = SqliteConnections.getPooledDataSource(
+                tempFolder.newFolder("randomFile3").toPath());
+
+        defaultLocalServer = createLogComponentsForServer(localDataSource);
+        defaultRemoteServerList = ImmutableList.of(
+                createLogComponentsForServer(remoteDataSource1), createLogComponentsForServer(remoteDataSource2));
+        paxosLogHistoryProvider = paxosLogHistoryProvider();
+    }
+
+    private PaxosLogHistoryProvider paxosLogHistoryProvider() {
+        return new PaxosLogHistoryProvider(
+                localDataSource,
+                defaultRemoteServerList.stream()
+                        .map(StateLogComponents::dataSource)
+                        .map(this::getHistoryProviderResource)
+                        .collect(Collectors.toList()));
+    }
+
+    public List<StateLogComponents> createStatLogComponentsForNamespaceAndUseCase(
```

Comment:
Same as elsewhere: this and other methods don't need to be public.

Reply:
also: `StatLog`?